tests/models/test_mtp.py

```diff
@@ -0,0 +1,285 @@
+    assert isinstance(energies, list)
+    assert energies == expected_energies
+
+
+@pytest.fixture
+def mock_extract_energy_from_thermo_log(mocker):
+    return mocker.patch('crystal_diffusion.train_mtp.extract_energy_from_thermo_log', return_value=[])
+
+
+@pytest.fixture
+def mock_extract_structure_and_forces(mocker):
+    return mocker.patch('crystal_diffusion.train_mtp.extract_structure_and_forces_from_file', return_value=([], []))
+
+
+def test_prepare_mtp_inputs_from_lammps(mock_extract_structure_and_forces, mock_extract_energy_from_thermo_log, tmpdir):
+    # Create mock file paths
+    output_yaml_files = [os.path.join(tmpdir, "output1.yaml"), os.path.join(tmpdir, "output2.yaml")]
+    thermo_yaml_files = [os.path.join(tmpdir, "thermo1.yaml"), os.path.join(tmpdir, "thermo2.yaml")]
+
+    # Mock atom dictionary
+    atom_dict = {1: 'H', 2: 'He'}
+
+    # Call the function
+    mtp_inputs = prepare_mtp_inputs_from_lammps(output_yaml_files, thermo_yaml_files, atom_dict)
+
+    # Verify that the mocks were called correctly
+    assert mock_extract_structure_and_forces.call_count == 2
+    mock_extract_structure_and_forces.assert_called_with(output_yaml_files[1], atom_dict)
+
+    assert mock_extract_energy_from_thermo_log.call_count == 2
+    mock_extract_energy_from_thermo_log.assert_called_with(thermo_yaml_files[1])
+
+    # Verify that the result is correctly structured
+    assert isinstance(mtp_inputs.structure, list)
+    assert isinstance(mtp_inputs.energy, list)
+    assert isinstance(mtp_inputs.forces, list)
+
+    # Verify that the data from the mocks is aggregated into the results correctly
+    assert mtp_inputs.structure == mock_extract_structure_and_forces.return_value[0] * len(output_yaml_files)
+    assert mtp_inputs.forces == mock_extract_structure_and_forces.return_value[1] * len(output_yaml_files)
+    assert mtp_inputs.energy == mock_extract_energy_from_thermo_log.return_value * len(thermo_yaml_files)
+
+
+def test_get_metrics_from_pred():
+    # test function from train_mtp
+    # TODO get better metrics and refactor the script
+    # Assuming there are 2 structures, each with 2 atoms (Total 4 readings)
+    df_orig = pd.DataFrame({
+        'structure_index': [0, 0, 1, 1],
+        'atom_index': [0, 1, 0, 1],
+        'energy': [1, 1, 3, 3],  # Total energy for the structure is the same for both atoms
+        'fx': [0.1, -0.1, 0.2, -0.2],
+        'fy': [0.1, -0.1, 0.2, -0.2],
+        'fz': [0.1, -0.1, 0.2, -0.2]
+    })
+
+    # Predicted data with some error introduced
+    df_predict = pd.DataFrame({
+        'structure_index': [0, 0, 1, 1],
+        'atom_index': [0, 1, 0, 1],
+        'energy': [1.05, 0.95, 3.1, 2.9],  # Energy has a slight variation
+        'fx': [0.12, -0.08, 0.23, -0.17],
+        'fy': [0.09, -0.11, 0.19, -0.21],
+        'fz': [0.11, -0.09, 0.18, -0.22]
+    })
+
+    # Calculate expected MAE for energy and forces
+    expected_mae_energy = mean_absolute_error(
```

Comment:
This is different from what is done in the function being tested. The function being tested computes the MAE on the energy per atom, whereas this computes the MAE on the total energy. The test still passes because the test data produces an MAE of zero for both possibilities.

Reply:
I suggest changing the test data to be more "random".